### PR TITLE
Expand - add support for invoking procedural macros on Windows

### PR DESCRIPTION
**Pull request by arlosi**
This change adds support for invoking procedural macros on Windows using Win32 APIs.

### CODE REVIEW

File: src/expand/proc_macro.cpp [hunk 1]
```diff
@@ -163,15 +163,15 @@ struct ProcMacroInv:
...
     HANDLE  child_handle;
     HANDLE  child_stdin;
     HANDLE  child_stdout;
 #else
     // POSIX
     pid_t   child_pid;  // Questionably needed
-     int    child_stdin;
-     int    child_stdout;
+    int    child_stdin;
+    int    child_stdout;
```
Comment (bjorn3) [inline src/expand/proc_macro.cpp]:
```suggestion
    int     child_stdout;
```

Comment (thepowersgang) [reply src/expand/proc_macro.cpp]: You're pointing out the change in indentation? I assume it was visual studio getting oppinionated.

**Comment by thepowersgang** [comment]
Thanks for this, a few quick comments (none of which are blockers for merging really...)

 - Indentation changes around the unix handling (pretty minor really) - not really the scope of the change.
- I don't like using brace initialisation for primitive like values (the `DWORD bytesWritten{};`), kinda perfer using assignment of a value.

Feel free to argue for the current style.

File: src/expand/proc_macro.cpp (hunk 1 above)
Comment (bjorn3) [reply src/expand/proc_macro.cpp]: Yes, with this change it is aligned again.

Comment (arlosi) [reply src/expand/proc_macro.cpp]: I reverted the whitespace changes

**Pull request merged**